docbrown/db/benches/common/mod.rs

```diff
@@ -104,6 +104,55 @@ pub fn run_ingestion_benchmarks<F>(
     );
 }
 
+fn times(n: usize) -> impl Iterator {
+    std::iter::repeat(()).take(n)
+}
+
+pub fn run_large_ingestion_benchmarks<F>(
+    group: &mut BenchmarkGroup<WallTime>,
+    mut make_graph: F,
+    parameter: Option<usize>,
+) where
+    F: FnMut() -> Graph,
+{
+    let mut times_gen = make_time_gen();
+    let mut time_sample = || times_gen.next().unwrap();
+
+    let updates = 1000;
+
+    bench (
+        group,
+        "1k fixed edge updates with varying time",
+        parameter,
+        |b: &mut Bencher| {
+            b.iter_batched_ref(
+                || (make_graph(), time_sample()),
```

Comment:
Here we are creating just 1 time sample and reusing it for all the updates. In line 132 you can do instead:
`g.add_edge(time_sample(), 0, 0, &vec![])`

The only problem with this is that we are measuring the time to generate a sample as well. I quickly created a benchmark locally and it seems to take around 10 ns each time we call `time_sample`. 

An alternative to avoid including in the routine the generation of a sample would be generating a vector of samples in the setup and then consuming it in the routine, something like this:

```rust
b.iter_batched_ref(
    || (make_graph(), make_time_gen().take(updates).collect::<Vec<i64>>()),
    |(g, times)|
        for t in times.iter() {
            g.add_edge(*t, 0, 0, &vec![])
        },
    BatchSize::SmallInput,
)
```
You can do something similar for the other bench as well.

If this starts getting tricky though, we can ignore it for the time being as we are going to catch performance regressions in any case.